src/perfEvents_linux.cpp

```diff
@@ -592,12 +593,12 @@ int PerfEvents::createForThread(int tid) {
 
     int fd;
     if (FdTransferClient::hasPeer()) {
-        fd = FdTransferClient::requestPerfFd(&tid, &attr);
+        fd = FdTransferClient::requestPerfFd(&tid, _target_cpu, &attr);
     } else {
-        fd = syscall(__NR_perf_event_open, &attr, tid, -1, -1, PERF_FLAG_FD_CLOEXEC);
+        fd = syscall(__NR_perf_event_open, &attr, tid, _target_cpu, -1, PERF_FLAG_FD_CLOEXEC);
```

Comment:
It is worth passing `args._target_cpu` argument to `perf_event_open` call in `check()` function too.
This is to ensure `check` fails whenever `start` fails with the same arguments.

Reply:
6004b68ce96e3585b1dceef606dfd1ce706cfbc6